scenario/scripts/snapshot.py

```diff
@@ -0,0 +1,752 @@
+    except InvalidModel:
+        logger.critical(f"invalid model: {model!r} not found.")
+        exit(1)
+
+    logger.info(f"snapshot done.")
+
+    if pprint:
+        if format == FormatOption.pytest:
+            charm_type_name = try_guess_charm_type_name()
+            txt = format_test_case(state, charm_type_name=charm_type_name)
+        elif format == FormatOption.state:
+            txt = format_state(state)
+        elif format == FormatOption.json:
+            txt = json.dumps(asdict(state), indent=2)
+        else:
+            raise ValueError(f"unknown format {format}")
+
+        print(txt)
+
+    return state
+
+
+def snapshot(
+    target: str = typer.Argument(..., help="Target unit."),
+    model: Optional[str] = typer.Option(
+        None, "-m", "--model", help="Which model to look at."
+    ),
+    format: FormatOption = typer.Option(
+        "state",
+        "-f",
+        "--format",
+        help="How to format the output. "
+        "``state``: Outputs a black-formatted repr() of the State object (if black is installed!). "
+        "``json``: Outputs a Jsonified State object. "
+        "``pytest``: Outputs a full-blown pytest scenario test based on this State. ",
+    ),
+    include: str = typer.Option(
+        "rckn",
+        "--include",
+        "-i",
+        help="What data to include in the state. "
+        "``r``: relation, ``c``: config, ``k``: containers ``n``: networks.",
+    ),
+    include_dead_relation_networks: bool = typer.Option(
+        False,
+        "--include-dead-relation-networks",
+        help="Whether to gather networks of inactive relation endpoints.",
+        is_flag=True,
+    ),
+    include_juju_relation_data: bool = typer.Option(
+        False,
+        "--include-juju-relation-data",
+        help="Whether to include in the relation data the default juju keys (egress-subnets,"
+        "ingress-address, private-address).",
+        is_flag=True,
+    ),
+) -> State:
+    """Gather and output the State of a remote target unit.
+
+    If black is available, the output will be piped through it for formatting.
+
+    Usage: snapshot myapp/0 > ./tests/scenario/case1.py
+    """
+    return _snapshot(
+        target=target,
+        model=model,
+        format=format,
+        include=include,
+        include_juju_relation_data=include_juju_relation_data,
+        include_dead_relation_networks=include_dead_relation_networks,
+    )
+
+
+if __name__ == "__main__":
+    print(
+        _snapshot(
+            "trfk/0",
+            model="foo",
+            format=FormatOption.json,
+            include="k",
+            fetch_files={
+                "traefik": [
+                    Path("/opt/traefik/juju/certificates.yaml"),
+                    Path("/opt/traefik/juju/certificate.cert"),
+                    Path("/opt/traefik/juju/certificate.key"),
+                    Path("/etc/traefik/traefik.yaml"),
+                ]
+            },
+        )
+    )
```

Comment:
Slightly confused here. This code is snapshotting a traefik charm, however, not everyone running snapshot is going to have a traefik unit deployed. Is this just testing code? If so, it should be removed before merging this pull request.

```suggestion
```